src/openrct2/actions/PlacePeepSpawnAction.hpp

```diff
@@ -100,17 +98,12 @@ DEFINE_GAME_ACTION(PlacePeepSpawnAction, GAME_COMMAND_PLACE_PEEP_SPAWN, GameActi
         res->ExpenditureType = RCT_EXPENDITURE_TYPE_LAND_PURCHASE;
         res->Position = CoordsXYZ{ _location.x, _location.y, _location.z / 8 };
 
-        // If we have reached our max peep spawns, use peep spawn next to last one set.
-        if (gPeepSpawns.size() >= MAX_PEEP_SPAWNS)
+        // If we have reached our max peep spawns, remove the oldest spawns
+        while (gPeepSpawns.size() >= MAX_PEEP_SPAWNS)
         {
-            auto peepSpawnIndex = _nextPeepSpawnIndex;
-            _nextPeepSpawnIndex = (peepSpawnIndex + 1) % (int32_t)gPeepSpawns.size();
-
-            // Before the new location is set, clear the old location
-            int32_t prevX = gPeepSpawns[peepSpawnIndex].x;
-            gPeepSpawns[peepSpawnIndex].x = PEEP_SPAWN_UNDEFINED;
-
-            map_invalidate_tile_full(prevX, gPeepSpawns[peepSpawnIndex].y);
+            auto oldestSpawn = gPeepSpawns.begin();
+            gPeepSpawns.erase(oldestSpawn);
+            map_invalidate_tile_full((*oldestSpawn).x, (*oldestSpawn).y);
```

Comment:
`vector::erase` invalidates all iterators at and after the given iterator. You need to store the coordinates before removing it.

Reply:
👍 made the fix.

Out of interest: this would also have been solved by swapping the `erase` and `invalidate` lines, but would that have been a safe thing to do? I noticed that `map_invalidate_tile_full` calls tend to be the last things functions do, but the conditions where that really matters are not entirely clear.

Reply:
In this case, the order doesn't really matter since it simply tells the game what screen areas to redraw on the next frame. Though invalidating them after having made the changes it the way to go, in case the invalidating process gets used by something else later on.

Reply:
Ah, that makes sense. Thanks for explaining!